src/Main.cpp

```diff
@@ -66,6 +68,16 @@
 #include "device.h"
 #endif
 
+// Check if on a GNU system (Linux) or other platform
+#if defined(__GNUC__) || defined(__linux__)
+#define ENABLE_FLOAT_EXCEPTIONS(rank)                                                              \
```

Comment:
It's best to move the `if` to line 116 IMO, to avoid the `ENABLE_FLOAT_EXCEPTIONS` as far as possible—macros can cause a lot of trouble and are hence best reduced to minimum.